go-controller/pkg/ovn/gateway.go

```
@@ -17,6 +17,35 @@ func (ovn *Controller) getOvnGateways() ([]string, error) {
 	return strings.Fields(string(out)), err
 }
 
+func (ovn *Controller) getGatewayPhysicalIP(physicalGateway string) string {
+	out, err := exec.Command(OvnNbctl, "get", "logical_router",
+		physicalGateway, "external_ids:physical_ip").CombinedOutput()
+	if err != nil {
+		logrus.Errorf("Error in getting gateway physical IP from %s: %v(%v)",
+			physicalGateway, string(out), err)
+		return ""
+	}
+
+	physicalIP := strings.Trim(string(out), "\" \n")
+	return physicalIP
+}
+
+func (ovn *Controller) getGatewayLoadBalancer(physicalGateway,
```

Same here, I'd just return an error on "" loadbalancer IP.

---

same comment.